### ISSUE TITLE
Discussion: Header and Footer Menu Items

### ISSUE THREAD

**Issue by pkscout** [xbmc/kodi-tv#157]
We're having bits and pieces of this conversation in two or three different issues, so I thought I'd get this all in one place.  So that there is something to which folks can react, I'm going to propose both a philosophy for the menus and a layout based on that.

## Proposed Philosophy
The header menu and it's sub menus should contain that which we want the visitor to see early and/or get to quickly. The footer should contain things people might want to know or that we need to include for regulatory reasons.  Header sub menus should contain no more than five items (four if the submenu includes a footer).  Footer sections should contain no more than nine items.  As much as possible, the header and footer menus should not duplicate each other.

## Proposed Menu Layout
Based on that, the first thing I think we should do is remove the social icons from the header. We seem to not use those social outlets much and don't seem overly concerned about making them more visible for visitors.  Beyond that:

### Header Menu

**About**
- Working on Kodi
- About Kodi
- Sponsors
- Software
- Contact

**News**

**Download**

**Add-ons**
- About Add-ons
- Matrix Add-ons
- Leia Add-ons
- Footer with note about selecting correct version

**Help**
- Wiki
- Developer Documentation
- Forum
- IRC
- Matrix.to

**Store**  
(I know this got renamed to Merch, but I really don't like the slang shortened English phrase)

**Donate**

### Footer

**About**
- Kodi Foundation
- Kodi Team
- Sponsorships
- Terms of Service
- Privacy Policy
- DMCA Policy
- GPL Violators

**Sponsors**
- (the current list of sponsors)

**Friends**
- Fanart.TV
- LibreELEC
- OSMC
- MusicBrainz.org
- TheAudioDB.com
- TheGamesDB.net
- TheMovieDB.org
- TVDB.com

**Social**
- Facebook
- Twitter
- YouTube
- GitHub
- RSS

**Comment by Darren-Hill** [comment]
Is Developer Documentation in the help menu the Doxygen stuff?

Generally the suggestion works fine for me, although I would tweak the header menu order a little and swap Help and Store over so Help is to the far right (between Store and Donate). Generally speaking the help menu is on the right hand side of menus and it feels a little odd to me to have it where it currently is (although having the Donate button on the right hand end works for me still as it's a distinct button).

One observation is we do seem to be giving our Sponsors a lot of webspace, with a page/header item, a banner on the main page and a list in the footer. Maybe drop the page banner?

**Comment by Darren-Hill** [comment]
Also we seem to have lost the header route to contribute with the link to the main GitHub page. Shouldn't that be rather more prominent somewhere in the header, along perhaps with a link to the bug tracker too similar to the one on the forum header?

Thinking here to try and get people to the repo to drive people towards contributions...

**Comment by pkscout** [comment]
> Is Developer Documentation in the help menu the Doxygen stuff?

Yes, the developer documentation is the Doxygen stuff.

> Also on the socials we have a few others, like LinkedIn and Mastedon (if that's still going)? Plus IRC of course if we can work out the linkage for it.

That's fine. For IRC (and I guess Matrix.to as well) we really need to decide if they are social things or support things and put them in the appropriate place.

> Also we seem to have lost the header route to contribute with the link to the main GitHub page. Shouldn't that be rather more prominent somewhere in the header, along perhaps with a link to the bug tracker too similar to the one on the forum header?

We have the contribute page, and perhaps we should expand that to include things like the bug tracker and the Github link. We can also add Github in the social footer (I had forgotten it was in the social icons).  If Matrix.to and/or IRC move out of the help menu, then we could put Working on Kodi back into the Help menu.

**Comment by Darren-Hill** [comment]
You haven't actually listed the contribute page in the original post above, nor is it currently linked in the menu on the staging branch :) . 

Just opened it up directly on the staging site and it's very much what I was thinking of, although it's a bit of an "in your face" full page. It may also be better there to have a banner or somesuch with a more prominent link to the GitHub repo, rather than just the one at the bottom of the developer sub-page?

**Comment by pkscout** [comment]
> You haven't actually listed the contribute page/header menu item in the original post above, nor is it currently linked in the menu on the staging branch :)

It's called **Working on Kodi** now, but is still the contribute page.

**Comment by Darren-Hill** [comment]
For me it should be a top level menu, not a sub-menu item. Having it under "About" also doesn't feel intuitively where I would expect to find it anyway (but none of the current top level items do either).

Let's take the opportunity to push it as much as we can and make it prominent and simple to find.

**Comment by pkscout** [comment]
I know there is this tendency to want to fix things by just putting more in the header, but this is the image I keep close to my heart when thinking about the header:

<img width="881" alt="Screen Shot 2021-04-04 at 10 45 30 AM" src="https://user-images.githubusercontent.com/1402018/113521089-0a15be80-9533-11eb-9272-e7797ed8786a.png">

That is what the header looks like at the very smallest screen before it shifts to mobile. If you want something like Contribute in the main menu, then you have to give up one (or probably two other things). Either that or we can sacrifice the entire tablet experience and make those folks use the mobile UI.  If we do that, then this is the space we have left before it would shift to mobile:

<img width="1132" alt="Screen Shot 2021-04-04 at 10 49 36 AM" src="https://user-images.githubusercontent.com/1402018/113521192-8b6d5100-9533-11eb-9f08-3c1b0f3fb29d.png">

**Comment by pkscout** [comment]
There are updates on staging with a new header based on these conversations. It includes a Contribute menu (which means the mobile interface is now used for tablets too), adds IRC and Matrix.to in the Help menu, and aligns the other menus and footer with the initial proposal. I also added another slide to the carousel with a  specific call for developers that links to the developer page directly (all three index page options have been updated with this slide).  The contribute page has been reordered, and I reworked the developer call page to show the Wiki and Github links earlier.  All the code for this is still sitting in a branch I have locally (i.e. I haven't merged it into staging).

https://kodi-tv-staging.netlify.app

**Comment by Darren-Hill** [comment]
Same on my 8" Android tablet, so it would seem to be reasonably generic. 

The new set-up looks a lot better to me, although I'm not sure the header order is still correct. Leading with "about" doesn't feel inuitively right to me - maybe move that to in-between contribute and store? But it's just fine-tuning, I'm happy with the content as-is now.

For me the main 3 things we want from the site are donations, downloads and contributions. Everything else is more for others benefit (users and sponsors mostly), so it makes sense to me that we should use headers/footers and the banner slides primarily for our own benefit

**Comment by Razzeee** [comment]
Seems sensible in general. Only things I'm wondering about are:

- Should friends & social be ordered by alphabet
- Should RSS really be under social? Where else?
- Should chats be under social?

**Comment by pkscout** [comment]
I alphabetized the social list.

RSS really has no where else to go. If we try to make it a menu item under News that is going to be strange. Many of the feed readers look for an RSS link on the home page so that when you put just the site address in it finds it, so I think having a spot in the footer under social is the easiest way to do this.

If IRC and Matrix.to are considered official support venues, they should be in the Help menu. If they are chat that just happen to have support happening in them, put them under social. I don't use either of them, so someone who does needs to decide.

**Comment by pkscout** [comment]
I've merged all the changes based on discussions above into staging.

**Comment by Darren-Hill** [comment]
I'll ask in the mods forum if anyone there does use IRC or Matrix.to for support purposes.

**Comment by Razzeee** [comment]
I'm aware of support happening both via matrix and irc.

**Comment by Darren-Hill** [comment]
So far there are a couple at least who do IRC (either directly or via Slack bridge).

No-one yet has indicated Matrix.to, but Razzeee knows more there it seems.

**Comment by Razzeee** [comment]
Yeah, matrix isn't done by moderators, but by team memebers/devs at the moment

**Comment by pkscout** [comment]
It feels like it's a toss up then for IRC and Matrix.to. I'd say if we want to promote them, then they should stay in the help menu. If we just want them to be there in case someone wants to, then put them in the footer.

**Comment by Razzeee** [comment]
Let's go for the help menu

**Comment by pkscout** [comment]
Sounds good to me, since that's where they are now. So no more work.  '-)

I'm going to label this needs testing, but I think all the changes we've discussed have been made.

**Comment by pkscout** [comment]
Can I go ahead and close this and https://github.com/xbmc/kodi-tv/issues/144? It feels like we've got consensus on the header and footer at this point (including the conversation about the IRC/Matrix.to links).  I'll let @Darren-Hill make a decision about https://github.com/xbmc/kodi-tv/issues/49 and https://github.com/xbmc/kodi-tv/issues/2. We at least have those pages linked now, it's a question of whether they are actually done.

**Comment by Razzeee** [comment]
Just went through staging again, and I'm a bit confused by "Developer resources" any ideas for a better naming, as it only shows Kodi api docs?

**Comment by pkscout** [comment]
It isn't just API docs though. docs.kodi.tv has skin development stuff too.  I don't know how complete it is, but I have the feeling the long term plan is to move all the development docs here so that the wiki is only user documentation.

**Comment by yol** [comment]
I think the menu items came out nicely, thanks!

Two nits:

The Matrix menu item should be named just "Matrix", not "Matrix.to" (and the text should read "[...] hanging out in Matrix") because that is the name of the service. The `matrix.to` domain is just a redirector to convenience so people are shown a choice of clients and we have something to link to.

The IRC text should be "Join us in #kodi on FreeNode or [...]" to be more natural. (Almost?) every IRC client has FreeNode already set up, so no need to spell out irc.freenode.net.

On another note, I still think our GitHub should be more prominently featured as it is our main development vehicle and putting it under "Social" in the footer doesn't feel quite right. But I guess there's no space left in a reasonable location?

**Comment by Razzeee** [comment]
Addressed in https://github.com/xbmc/kodi-tv/pull/164 minus the GitHub problem.

**Comment by pkscout** [comment]
Here are two options for more prominent Github placement.

In the main header (there is just enough room for one last item):
<img width="1141" alt="Screen Shot 2021-04-10 at 6 22 01 AM" src="https://user-images.githubusercontent.com/1402018/114277242-1f249e80-99c6-11eb-9f24-d52efc1dc714.png">

As a submenu of Contribute:
<img width="1504" alt="Screen Shot 2021-04-10 at 6 28 37 AM" src="https://user-images.githubusercontent.com/1402018/114277249-2b106080-99c6-11eb-9fa7-4a3c66fc7d76.png">

**Comment by Razzeee** [comment]
I like the first one

**Comment by pkscout** [comment]
Added Github to main menu (and cleaned up the header a bit). Building to the staging site now.

**Comment by Darren-Hill** [comment]
One other thought - how would a GitHub button look, similar to the donate one?

Maybe positioned on the left of the menu between Kodi and News, to balance it?

**Comment by pkscout** [comment]
It would look like this:
<img width="1504" alt="Screen Shot 2021-04-11 at 7 04 40 AM" src="https://user-images.githubusercontent.com/1402018/114313907-3cc03980-9a94-11eb-877d-d3b3a81dd143.png">

**Comment by Darren-Hill** [comment]
I quite like that. What do the rest of you think?

**Comment by Razzeee** [comment]
I think that's a no go. As both buttons are primary buttons and fighting for your attention, which is extra hard if they're not next to each other. It also highlights how the menu strip is not centered and throws off the design.

Here's an approach, that's balanced, but probably makes mobile layouts harder
![image](https://user-images.githubusercontent.com/5943908/114316592-87799a00-9b04-11eb-8ae3-ea5a120b53ce.png)

**Comment by pkscout** [comment]
We could probably re-use the logic from the social menu for those two buttons. I'll have to track that down in the commits, as I just got rid of the last of the social stuff this morning, but it's doable. I do wonder if having the "important" buttons all the way to the right is a good idea though. That's one of the last places the eye looks when scanning a page.

We could also put both highlighted buttons together at the beginning or end of the menu (which would be easier programmatically).

**Comment by Razzeee** [comment]
We might also just be off the mark, while I get wanting that button there, is it really important enough? I have certainly been on websites of oss projects and was annoyed by not being able to find their code, but mostly those are not end user facing projects.

But we're end user facing. Projects like Gnome and VLC seem to not have their code links front and center.
https://www.videolan.org/
https://www.gnome.org/

**Comment by Darren-Hill** [comment]
I just come back to the same basic thought from before - what do we actually want from the site?

For me, we want downloads, donations and access to the code. Everything else is secondary, and/or primary for others.

Thus for me the header should reflect that.

**Comment by Razzeee** [comment]
I think developers will want code access, but even those are probably not always looking for the code. I think that might be 10% of site users top, and I think that's generous.

**Comment by Razzeee** [comment]
And while asking what we want is nice, but it would be better to ask what users want.

**Comment by Darren-Hill** [comment]
Users want those three items, plus information.

But we can't highlight everything, so we highlight what is best for us.

We should make it easy for people to give us money (donations), code (GitHub) and to get the software (Downloads)

**Comment by pkscout** [comment]
Here's a couple more takes. I've added a third button type (so we have primary, secondary, and regular) and moved stuff around.

Two primary buttons, one secondary:
<img width="1504" alt="Screen Shot 2021-04-11 at 10 14 12 AM" src="https://user-images.githubusercontent.com/1402018/114319859-fb895300-9aae-11eb-9341-b84f13914ce3.png">

One primary button, two secondary:
<img width="1504" alt="Screen Shot 2021-04-11 at 10 14 41 AM" src="https://user-images.githubusercontent.com/1402018/114319881-0ba13280-9aaf-11eb-9cd7-126656f95fb8.png">

**Comment by Darren-Hill** [comment]
For me they look better on the right hand side - all on the left with the Kodi icon too makes everything seem left-heavy and lopsided to me, and menu just kind of trails off a bit. At least with the Kodi icon/name on the left and the buttons on the right then everything is bookended, but I take the point made above about where the visual focus ends up.

Also the two different button styles for me don't work well. They almost look more like a coding error than a deliberate style decision. I really much prefer them to be all the same style.

**Comment by pkscout** [comment]
I think we are at the point of finding the option that the largest number of people hate the least. I might be able to handle two of the brighter primary buttons, but not three.  I'd rather have our important things on the far left, but I definitely don't want them out on their own where the social icons were.  So with that in mind, here's the option I think I hate the least:

<img width="1504" alt="Screen Shot 2021-04-11 at 11 27 57 AM" src="https://user-images.githubusercontent.com/1402018/114321887-00530480-9ab9-11eb-891d-f374d4b6af1a.png">

Basically, I didn't get anything I wanted, but neither did anyone else. Pretty sure that's the definition of a perfect compromise.

'-)

**Comment by yol** [comment]
How do people feel about using the GitHub icon instead of the Text? Like the earlier social buttons?
-- 
K-9 Mail で Android デバイスから送信しました。簡単で申し訳ありません。

**Comment by Darren-Hill** [comment]
@pkscout I could also certainly live with the last design posted above.

Fully agree that this is going to be a least-worst type scenario as it's very much personal taste. Not keen on the icon thought, would look odd in an otherwise fully text header (aside from the Kodi logo).

**Comment by Razzeee** [comment]
Closing this for now, staging seems fine. 

This issue is getting too long too, so feel free to create new issues, if anything still bugs you.
